nexus/db-queries/src/db/datastore/lldp.rs

```diff
@@ -0,0 +1,178 @@
+use super::DataStore;
+use crate::context::OpContext;
+use crate::db;
+use crate::db::error::public_error_from_diesel;
+use crate::db::error::ErrorHandler;
+use crate::db::model::LldpLinkConfig;
+use async_bb8_diesel::AsyncRunQueryDsl;
+use chrono::Utc;
+use diesel::ExpressionMethods;
+use diesel::QueryDsl;
+use diesel::SelectableHelper;
+use ipnetwork::IpNetwork;
+use omicron_common::api::external;
+use omicron_common::api::external::Error;
+use omicron_common::api::external::LookupResult;
+use omicron_common::api::external::Name;
+use omicron_common::api::external::ResourceType;
+use omicron_common::api::external::UpdateResult;
+use uuid::Uuid;
+
+// The LLDP configuration has been defined as a leaf of the switch-port-settings
+// tree, and is identified in the database with a UUID stored in that tree.
+// Using the uuid as the target argument for the config operations would be
+// reasonable, and similar in spirit to the link configuration operations.
+//
+// On the other hand, the neighbors are discovered on a configured link, but the
+// data is otherwise completely independent of the configuration.  Furthermore,
+// the questions answered by the neighbor information have to do with the
+// physical connections between the Oxide rack and the upstream, datacenter
+// switch.  Accordingly, it seems more appropriate to use the physical
+// rack/switch/port triple to identify the port of interest for the neighbors
+// query.
+//
+// For consistency across the lldp operations, all use rack/switch/port rather
+// than the uuid.
+// XXX: Is this the right call?  The other options being: uuid for all
```

Comment:
Is it? Should we decide before merging this?

Reply:
I think it is, which is why I implemented it this way.  The comment was meant to solicit feedback from the reviewers with more API experience - _i.e_, you.  I probably should have made that a github/review comment rather than a rust comment to make that clearer.

Reply:
Thanks for clarifying. I think it's fine to use rack_id / switch-port rather than the uuid. I do think it's a little odd that the port is often a path param and the rack is a query param. Was there some particular motivation for that?

Reply:
I would suggest using both the rack id and port in the path, and there seems to be at least some precedent for that.